x-pack/plugins/cloud_integrations/cloud_chat/server/config.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { get, has } from 'lodash';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { PluginConfigDescriptor } from '@kbn/core/server';
+
+const configSchema = schema.object({
+  enabled: schema.boolean({ defaultValue: false }),
+  chatURL: schema.maybe(schema.string()),
+  chatIdentitySecret: schema.maybe(schema.string()),
+});
+
+export type CloudChatConfigType = TypeOf<typeof configSchema>;
+
+export const config: PluginConfigDescriptor<CloudChatConfigType> = {
+  exposeToBrowser: {
+    chatURL: true,
+  },
+  schema: configSchema,
+  deprecations: () => [
+    // Silently move the chat configuration from `xpack.cloud` to `xpack.cloud_integrations.chat`.
+    // No need to emit a deprecation log because it's an internal restructure
+    (cfg) => {
+      return {
+        set: [
+          ...copyIfExists({
+            cfg,
+            fromKey: 'xpack.cloud.chat.enabled',
+            toKey: 'xpack.cloud_integrations.chat.enabled',
+          }),
+          ...copyIfExists({
+            cfg,
+            fromKey: 'xpack.cloud.chat.chatURL',
+            toKey: 'xpack.cloud_integrations.chat.chatURL',
+          }),
+          ...copyIfExists({
+            cfg,
+            fromKey: 'xpack.cloud.chatIdentitySecret',
+            toKey: 'xpack.cloud_integrations.chat.chatIdentitySecret',
+          }),
```

Comment:
I didn't know about the `{ silent: true }` option. I'll change it

Reply:
hmmm, I'm afraid we can't use it because UA will show the deprecation messages:

The `silent` deprecations are shown in the deprecation service: it uses [`configService.getHandledDeprecatedConfigs()`](https://github.com/elastic/kibana/blob/980e80b9021cf87a7609acffe2eceeb4adec1b42/packages/core/deprecations/core-deprecations-server-internal/src/deprecations_service.ts#L111) that returns the content of [`handledDeprecatedConfigs`](https://github.com/elastic/kibana/blob/1cbf83f73c5571d329086ee99fc30f62c5e690fc/packages/kbn-config/src/config_service.ts#L118-L120). And we always add every deprecation to that `Map` in [`markDeprecatedConfigAsHandled`](https://github.com/elastic/kibana/blob/1cbf83f73c5571d329086ee99fc30f62c5e690fc/packages/kbn-config/src/config_service.ts#L220-L225) called here.

We can revisit this implementation later on and adapt this piece of code if needed.

Reply:
Thanks for checking